### ISSUE TITLE
Sync stopped after ownCloud runs out of disc space, even if old files were deleted

### ISSUE THREAD

**Issue by jfqd** [owncloud/client#5346]
If you sync a Mac folder with the Sync-Client to the ownCloud and the ownCloud runs out of disc space, the sync is stopped and the remaining files are marked with a yellow dot in in the macOS Finder folder.

As far as well. When you now empty space by deleting old files on the macOS sync-folder, the files are also deleted in the ownCloud folder on the Server and the free disc space is recalculated and updated.

BUT the remaining files aren’t synced - even if you pause and restart the process. ONLY if you rename the parent folder, the remaining files are synced.

### Expected behaviour

After freeing disk space on the ownCloud the  remaining files should be synced again.

### Actual behaviour

Remaining files aren’t synced - even if you pause and restart the process.

### Steps to reproduce

1. Fill up your ownCloud Disk space
2. Try to sync additional files
3. Remove other files locally to free ownCloud Disk space
4. Skipped Files remain not be synced

### Server configuration
ownCloud version: 9.0.5

### Client configuration
Client version: 2.2.4

Operating system: macOS 10.12

Installation path of client: /Applications/owncloud.app

**Comment by SamuAlfageme** [comment]
@jfqd I can reproduce this. The file is blacklisted because of insufficient storage in the server stays like that until you modify its name/path (e.g. as you pointed changing the parent folder name).

The error that we display now when an upload fails due to file size being bigger than the remaining space available also could be improved:

```
The item is not synced because of previous errors: Error downloading https://XXXX/remote.php/webdav/60M-chunking-1673196607-7-1 - server replied: Insufficient Storage (Sabre\DAV\Exception\InsufficientStorage)
```

**Comment by SamuAlfageme** [comment]
Just encountered with another scenario that reproduces this condition: if an account has a quota N and reaches it, any additional files the sync client tries to upload are blacklisted as server throws the `Exception\InsufficientStorage`, If the admin then widens the quota to N+m, the sync keeps blacklisting those extra items even if the whole local folder space in disk is < N+m. 

tl;dr: some items in the blacklist should be updated upon changes in the quota/storage used in the server.

**Comment by ckamm** [comment]
@SamuAlfageme Thanks for the analysis. All blacklisted files will be retried, so the files should be uploaded eventually. The blacklist times are approximately 25s, 2min, 10min, 1h, 5h, 24h - so if the file has already been unsuccessfully tried 6 times, you'll have to wait for a day for the next attempt.

That said, I'm not convinced it's correct to temporarily blacklist files with insufficent-storage errors. This is a problem that users should be made aware off because they probably need to do something about it. @ogoffart @guruz @jturcotte Do you see a compelling reason to blacklist these errors?

**Comment by SamuAlfageme** [comment]
@ckamm oh, those blacklist retry times seem quite exponential 😄 (don't know how time-consuming this operation can be. but 24h seem quite extreme).

I think blacklist could do fine for this specific issue as long as it's handled wisely: **reviewing** the BL upon local changes for the first scenario, and based on a quota change (that the client notices almost immediately) for the second:

![quota_change](https://cloud.githubusercontent.com/assets/2644445/21177409/6404c096-c1eb-11e6-857f-eda431803617.gif)

**Comment by guruz** [comment]
Yes, we can look for the string ```InsufficientStorage``` and then behave differently

**Comment by ckamm** [comment]
@guruz Pull request at https://github.com/owncloud/client/pull/5424

**Comment by guruz** [comment]
@jfqd Do you want to help out testing this? http://download.owncloud.com/desktop/daily/ownCloud-2.3.0.3936-nightly20170105.pkg

**Comment by ogoffart** [comment]
I think maybe the fix is a bit extreme.  We still want to blacklist insuficient storage if the users did not do anything to solve the problem  (removing files or asking the admin)
Otherwise the client will continue to upload the file over and over again, wasting bandwidth.

I suggest instead to limit the blacklist fo 5h max. Or try to be smarter and detect the Insufficient Storage before it happens with the quota check. (like we do already for the hardrive space on the client)

**Comment by SamuAlfageme** [comment]
Working as expected, not blacklisted and when size is freed on the server the sync starts. 

However, in the second scenario, when there's a quota modification (like in https://github.com/owncloud/client/issues/5346#issuecomment-264810401) the sync is not rescheduled (we could easily `Force sync now` on detection of this event, since this does the magic)

**Comment by ckamm** [comment]
Patch was reverted. Bad behavior will be adressed as part of https://github.com/owncloud/client/issues/5537
